src/Analyzers/CSharp/Tests/UseIsNullCheck/UseIsNullCheckForReferenceEqualsTests.cs

```diff
@@ -500,5 +501,67 @@ void M(string s2)
     }
 }");
         }
+
+        [WorkItem(23581, "https://github.com/dotnet/roslyn/issues/47972")]
+        [Fact, Trait(Traits.Feature, Traits.Features.CodeActionsUseIsNullCheck)]
+        public async Task TestValueParameterTypeIsBaseConstraintGeneric()
+        {
+            await TestInRegularAndScript1Async(
+@"using System;
+
+class C
+{
+    public static void NotNull<T>(T value) where T:C
```

Comment:
this doesn't appear to be a value-typeparameter.  i.e. it doesn't have the 'struct' constraint.

Reply:
The name is unclear. The purpose is "no `class` constraint, but has base type (class) constraint"